apps/dashboard/forms.py

```diff
@@ -324,3 +326,71 @@ def save(self, commit=True):
                 self.instance.moderators.add(
                     *self.cleaned_data['add_moderators']
                 )
+
+
+class ExternalProjectBaseForm(forms.ModelForm):
+
+    start_date = forms.DateTimeField(required=False)
+    end_date = forms.DateTimeField(required=False)
+
+    class Meta:
+        model = extproject_models.ExternalProject
+        fields = ['name', 'url', 'description', 'image', 'is_archived'] \
+            + ['start_date', 'end_date']
+
+
+class ExternalProjectCreateForm(ExternalProjectBaseForm):
+
+    def __init__(self, view, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.view = view
```

Comment:
As discussed, its probably better to pass all variables required by the form more explicitly via get_form_kwargs.